bokehjs/src/lib/models/formatters/log_tick_formatter.ts

```diff
@@ -43,11 +45,19 @@ export class LogTickFormatter extends TickFormatter {
       return []
 
     const base = this.ticker != null ? this.ticker.base : 10
+    
+    const translate: { [id: string]: string; } = { "0":"⁰", "1":"¹", "2":"²", "3":"³", "4":"⁴", "5":"⁵", "6":"⁶", "7":"⁷", "8":"⁸", "9":"⁹", "+":"⁺", "-":"⁻" }
 
     let small_interval = false
     const labels: string[] = new Array(ticks.length)
     for (let i = 0, end = ticks.length; i < end; i++) {
-      labels[i] = `${base}^${ Math.round(Math.log(ticks[i]) / Math.log(base)) }`
+	  
+      if (!this.use_superscript_exponents) {
+		labels[i] = `${base}^${Math.round(Math.log(ticks[i]) / Math.log(base))}`
+	  } else {
+	    labels[i] = `${base}${(Math.round(Math.log(ticks[i]) / Math.log(base))).toString().replace(/[0-9\+\-]/g,  function(match){return translate[match]})}`
+	  }
+	  
```

Comment:
Whitespace should be cleaned up and redundant curly brackets removed. `Math.round(Math.log(ticks[i]) / Math.log(base))` should be extracted into a variable. Use arrow function instead of `function() {}`.

Reply:
Thank you for your suggestions. I am not an experienced JavaScript developer so your suggestions were very useful to me. I applied the suggested changes.